### PR TITLE
Remove default value handling from DefaultNotificationAccessor

### PR DESCRIPTION
**Pull request by JakeMathews**
# Remove default value handling from DefaultNotificationAccessor

**Link to github issue (if applicable):**
Fixes IALERT-2008

**If nothing above, what is your reason for this pull request:**
In the method `DefaultNotificationAccessor::getPageRequestForNotifications` the paging parameters are `@Nullable`, this should not be the case and default value handling should happen at a higher level.

**Changes proposed in this pull request:**
- Move default value handling higher into the API
- Change default page size from `Integer.MAX_VALUE` to `10`. Uses the constant defined in `AuditEntryPageModel`.

### CODE REVIEW

File: component/src/main/java/com/synopsys/integration/alert/component/audit/web/AuditEntryActions.java [hunk 1]
```diff
@@ -91,7 +92,9 @@ public AuditEntryActions(
         if (!authorizationManager.hasReadPermission(ConfigContextEnum.GLOBAL, descriptorKey)) {
             return new ActionResponse<>(HttpStatus.FORBIDDEN, ActionResponse.FORBIDDEN_MESSAGE);
         }
-        AuditEntryPageModel pagedRestModel = auditAccessor.getPageOfAuditEntries(pageNumber, pageSize, searchTerm, sortField, sortOrder, onlyShowSentNotifications, auditAccessor::convertToAuditEntryModelFromNotification);
+        Integer page = ObjectUtils.defaultIfNull(pageNumber, 0);
+        Integer size = ObjectUtils.defaultIfNull(pageSize, 100);
```
Comment (gkillough) [inline component/src/main/java/com/synopsys/integration/alert/component/audit/web/AuditEntryActions.java:96]: I think the class that `AuditEntryPageModel` extends might have constants that could be used for the default values.

**Review by gkillough — APPROVED** [review]

**Review by jamesrichard91 — APPROVED** [review]

File: component/src/main/java/com/synopsys/integration/alert/component/audit/web/AuditEntryActions.java (hunk 1 above)
Comment (bamandel) [reply component/src/main/java/com/synopsys/integration/alert/component/audit/web/AuditEntryActions.java:96]: If those constants don't exist we should at least make some in this class

**Review by bamandel — APPROVED** [review]

**Review by ChomickiM — APPROVED** [review]

**Review by psantos1113 — APPROVED** [review]

File: alert-common/src/main/java/com/synopsys/integration/alert/common/rest/model/AlertPagedModel.java [hunk 2]
```diff
@@ -27,8 +27,10 @@
 import net.minidev.json.annotate.JsonIgnore;
 
 public class AlertPagedModel<M extends AlertSerializableModel> extends AlertSerializableModel {
-    public static final String DEFAULT_PAGE_NUMBER = "0";
-    public static final String DEFAULT_PAGE_SIZE = "10";
+    public static final Integer DEFAULT_PAGE_NUMBER = 0;
+    public static final Integer DEFAULT_PAGE_SIZE = 10;
+    public static final String DEFAULT_PAGE_NUMBER_STRING = "0";
+    public static final String DEFAULT_PAGE_SIZE_STRING = "10";
```
Comment (JakeMathews) [inline alert-common/src/main/java/com/synopsys/integration/alert/common/rest/model/AlertPagedModel.java:33]: Spring requires a String constant in `ReadPageController`. We want to use the same defaults, but as an Integer. Should we convert the String to an Integer at runtime? Do what I've done here? Something else?

File: component/src/main/java/com/synopsys/integration/alert/component/audit/web/AuditEntryActions.java (hunk 1 above)
Comment (JakeMathews) [reply component/src/main/java/com/synopsys/integration/alert/component/audit/web/AuditEntryActions.java]: I am now using the constants in AlertPagedModel. But that introduces a new issue https://github.com/blackducksoftware/blackduck-alert/pull/1249#discussion_r520829142

File: alert-common/src/main/java/com/synopsys/integration/alert/common/rest/model/AlertPagedModel.java (hunk 2 above)
Comment (gkillough) [reply alert-common/src/main/java/com/synopsys/integration/alert/common/rest/model/AlertPagedModel.java:33]:
```java
    public static final Integer DEFAULT_PAGE_NUMBER = 0;
    public static final Integer DEFAULT_PAGE_SIZE = 10;
    public static final String DEFAULT_PAGE_NUMBER_STRING = DEFAULT_PAGE_NUMBER.toString();
    public static final String DEFAULT_PAGE_SIZE_STRING = DEFAULT_PAGE_SIZE.toString();
```
based on https://stackoverflow.com/a/654068/6921621

Comment (JakeMathews) [reply alert-common/src/main/java/com/synopsys/integration/alert/common/rest/model/AlertPagedModel.java:33]: Then it is no longer a constant. 
Screen Shot 2020-11-10 at 3 17 12 PM

Comment (gkillough) [reply alert-common/src/main/java/com/synopsys/integration/alert/common/rest/model/AlertPagedModel.java:33]: Alright, in that case maybe move the string constants into that interface and leave them as literals. Leave the integers in the paged model.

Comment (jamesrichard91) [reply alert-common/src/main/java/com/synopsys/integration/alert/common/rest/model/AlertPagedModel.java]: You could reverse it, leave the static Strings as literals and have the static integers based on those>
```
public static final Integer DEFAULT_PAGE_NUMBER = Integer.valueOf(ReadPageController. DEFAULT_PAGE_NUMBER);
public static final Integer DEFAULT_PAGE_SIZE =  Integer.valueOf(ReadPageController. DEFAULT_PAGE_SIZE);
```

Comment (JakeMathews) [reply alert-common/src/main/java/com/synopsys/integration/alert/common/rest/model/AlertPagedModel.java]: I was hesitant to do the String parsing at runtime, but I think that's better than declaring the constants twice.

**Pull request merged**